db/migrate/20230311033148_create_chapters.rb

```diff
@@ -0,0 +1,17 @@
+class CreateChapters < ActiveRecord::Migration[7.0]
+  def change
+    create_table :chapters do |t|
+      t.references :book, :foreign_key => true, :null => false
+      t.integer :number, :null => false
```

Comment:
We should probably also make it so that you don't create MULTIPLE chapters with the same number. Check if there's a way we can do this. 

Check if it's possible in a migration to guarantee that a column is unique or something. If not, we can just do it in the application side (like in the `Chapter` model).

Reply:
I'm not sure where to add the unique constraint for the chapter number. I'm sure the code is as follows :
add_index :chapters, :number unique: true
but I'm not sure where in the code it goes.


![](https://user-images.githubusercontent.com/99364516/224465949-b52b31b2-4477-456f-ad36-b24ddd4fae3b.png)

Reply:
<img width="581" alt="Screenshot 2023-03-11 at 12 16 51 AM" src="https://user-images.githubusercontent.com/99364516/224466410-7594231c-41f5-4c9e-a75c-a3e6ced8548b.png">


I believe I've resolved it at line 17 in the code. Please take a look.

Reply:
Awesome job. That is right! ✅ 

However, we have to take this one step further. 

The current behaviour will prevent ANY chapter from having the same `number`...  This is good for `book_1`, but when we bring `book_2` into the equation, that book is going to have to start from chapter number 1. 

Therefore, we need to add the unique constraint in a way that validates that each book has a unique chapter number. 

To do this, we can do:

`add_index :chapters, [:book_id, :number], :unique => true`

Reply:
<img width="1218" alt="Screenshot 2023-03-12 at 4 37 10 PM" src="https://user-images.githubusercontent.com/99364516/224572160-8c70964a-4e18-4d26-a89d-c694b3ab16f6.png">

I added the updated line, then I rollbacked and migrated the db again but it seems the unique constraint wasn't added to the schema, and its not working when I add the queries.

Reply:
<img width="1055" alt="Screen Shot 2023-03-12 at 5 43 08 PM" src="https://user-images.githubusercontent.com/17516559/224575486-b79c6936-123c-43dd-815e-dc21129d5bc3.png">

It should be working! It looks good on mind :)